lms/djangoapps/grades/rest_api/v1/views.py

```diff
@@ -207,3 +217,307 @@ def _get_course(self, request, course_id):
+                error_code='invalid_course_key'
+            )
+        if (GlobalStaff().has_user(request.user) or self.has_api_key_permissions(request)):
+            course_enrollments = CourseEnrollment.objects.select_related('user').filter(
+                is_active=True
+            )
+            if course_id:
+                course_enrollments = course_enrollments.filter(
+                    course_id=course_id
+                ).order_by('created')
+
+            if not all_users:
+                course_enrollments = course_enrollments.filter(user__username=username)
+
+            course_xblock_structure = self. _generate_course_xblock_structure(course_enrollments)
+            for course_key, course_info in course_xblock_structure.items():
+                course_data = self._get_course_data(
+                    course_key,
+                    course_info.get('course_enrollment'),
+                    course_info.get('course'),
+                    course_info.get('blocks')
+                )
+                data.append(course_data)
+            return Response(data)
+        else:
+            raise self.api_error(
+                status_code=status.HTTP_403_FORBIDDEN,
+                developer_message='The course does not exist.',
+                error_code='user_or_course_does_not_exist',
+            )
+
+    def _generate_course_xblock_structure(self, course_enrollments):
+        """ Generate a map of course to course enrollment and problem
+            xblocks for each of the course.
+        """
+        course_enrollment_id_map = {}
+        course_xblock_structure = {}
+        for course_enrollment in course_enrollments:
+            course_enrollment_id_map[str(course_enrollment.course_id)] = course_enrollment
+        for course_key, course_enrollment in course_enrollment_id_map.items():
+            course = get_course(course_enrollment.course_id, depth=4)
+            blocks = self._get_problem_blocks(course)
+            course_xblock_structure[course_key] = {
+                'course_enrollment': course_enrollment,
+                'blocks': blocks,
+                'course': course
+            }
+        return course_xblock_structure
+
+    def _get_problem_blocks(self, course):
+        """ Get a list of problem xblock for the course.
+        """
+        blocks = []
+        for section in course.get_children():
+            for subsection in section.get_children():
+                for vertical in subsection.get_children():
+                    for block in vertical.get_children():
+                        if (block.category == 'problem' and getattr(block, 'has_score', False)):
+                            blocks.append(block)
+        return blocks
+
+    def _get_problem_data(self, course_enrollment, block):
+        """
+        Get problem data from a course enrollment.
+
+        Args:
+        -----
+        course_enrollment: Course Enrollment.
+        block: Xblock to analyze.
```

Comment:
```suggestion
        block: XBlock to analyze.
```

Tiny nit: please make the naming consistent.